deepchecks/nlp/dataset.py

```diff
@@ -0,0 +1,557 @@
+        List of names for the feature columns in the DataFrame.
+    cat_features : t.Optional[t.Sequence[Hashable]] , default: None
+        List of names for the categorical features in the DataFrame. In order to disable categorical.
+        features inference, pass cat_features=[]
+    index_name : t.Optional[Hashable] , default: None
+        Name of the index column in the dataframe. If set_index_from_dataframe_index is True and index_name
+        is not None, index will be created from the dataframe index level with the given name. If index levels
+        have no names, an int must be used to select the appropriate level by order.
+    set_index_from_dataframe_index : bool , default: False
+        If set to true, index will be created from the dataframe index instead of dataframe columns (default).
+        If index_name is None, first level of the index will be used in case of a multilevel index.
+    datetime_name : t.Optional[Hashable] , default: None
+        Name of the datetime column in the dataframe. If set_datetime_from_dataframe_index is True and datetime_name
+        is not None, date will be created from the dataframe index level with the given name. If index levels
+        have no names, an int must be used to select the appropriate level by order.
+    set_datetime_from_dataframe_index : bool , default: False
+        If set to true, date will be created from the dataframe index instead of dataframe columns (default).
+        If datetime_name is None, first level of the index will be used in case of a multilevel index.
+    convert_datetime : bool , default: True
+        If set to true, date will be converted to datetime using pandas.to_datetime.
+    datetime_args : t.Optional[t.Dict] , default: None
+        pandas.to_datetime args used for conversion of the datetime column.
+        (look at https://pandas.pydata.org/docs/reference/api/pandas.to_datetime.html for more documentation)
+    max_categorical_ratio : float , default: 0.01
+        The max ratio of unique values in a column in order for it to be inferred as a
+        categorical feature.
+    max_categories : int , default: None
+        The maximum number of categories in a column in order for it to be inferred as a categorical
+        feature. if None, uses is_categorical default inference mechanism.
+    label_type : str , default: None
+        Used to assume target model type if not found on model. Values ('classification_label', 'regression_label')
+        If None then label type is inferred from label using is_categorical logic.
+    """
+
+    _dataset: datasets.Dataset
+    _task_type: t.Optional[TaskType]
+    _has_label: bool
+    _label_name: str
+    _is_multilabel: bool = False
+
+    def __init__(
+            self,
+            raw_text: t.Sequence[str],
+            label: t.Optional[TNLPLabel] = None,
+            task_type: str = None,
+            label_name: t.Optional[str] = None,
```

Comment:
why label name is required?

Reply:
For display purposes, may remove that if I find it's not needed

Reply:
Label name can be determined by task type, less arguments the better :)

Reply:
It's not the type, it's what it represents. "News topic", "named entity". "Medicine name" etc